integration-tests/tests/test_eth_api.py

```diff
@@ -0,0 +1,55 @@
+import random
+import pytest
+import jsonschema
+import json
+from conftest import option
+from test_cases import EthApiTestCase
+
+
+@pytest.mark.rpc
+@pytest.mark.ethclient
+class TestRpc(EthApiTestCase):
+    @pytest.mark.parametrize(
+        "iterations", range(3) # Run multiple times, each with a different Tx
+    )
+
+    def test_(self, iterations):
+        _id = str(random.randint(1, 8888))
+
+        __ = self.rpc_valid_request("ethclient_blockNumber", [self.network_id], _id)
+
+        __ = self.rpc_valid_request("ethclient_suggestGasPrice", [self.network_id], _id)
+
+        response = self.rpc_valid_request("ethclient_headerByNumber", [self.network_id, self.block_number], _id)
+        header = response.json()["result"]
+        assert header["number"] == self.block_number
+        assert header["hash"] == self.block_hash
+
+        response = self.rpc_valid_request("ethclient_blockByNumber", [self.network_id, self.block_number], _id)
+        block = response.json()["result"]
+        assert block["header"]["number"] == self.block_number
+        assert block["header"]["hash"] == self.block_hash
+        tx_hashes = [tx["hash"] for tx in block["transactions"]]
+        assert self.tx_hash in tx_hashes
+
+        response = self.rpc_valid_request("ethclient_headerByHash", [self.network_id, self.block_hash], _id)
+        header = response.json()["result"]
+        assert header["number"] == self.block_number
+        assert header["hash"] == self.block_hash
+
+        response = self.rpc_valid_request("ethclient_blockByHash", [self.network_id, self.block_hash], _id)
+        block = response.json()["result"]
+        assert block["header"]["number"] == self.block_number
+        assert block["header"]["hash"] == self.block_hash
+        tx_hashes = [tx["hash"] for tx in block["transactions"]]
+        assert self.tx_hash in tx_hashes
+
+        response = self.rpc_valid_request("ethclient_transactionByHash", [self.network_id, self.tx_hash], _id)
+        tx = response.json()["result"]
+        assert tx["tx"]["hash"] == self.tx_hash
+
+        response = self.rpc_valid_request("ethclient_transactionReceipt", [self.network_id, self.tx_hash], _id)
+        receipt = response.json()["result"]
+        assert receipt["transactionHash"] == self.tx_hash
+        assert receipt["blockNumber"] == self.block_number
+        assert receipt["blockHash"] == self.block_hash
```

Comment:
Hmm so in this test I want to make sure I'm getting consistent info about the same block/transaction when accessing the same piece of info from different status-go functions. If I split this into multiple tests I can no longer use `TransactionTestCase` as base, since it triggers a new Tx and stores a new tx_hash on every `setup_method`.

I can see the duplication though, would moving the asserts to functions help?

Reply:
@yevh-berdnyk @antdanchenko please check if the current form is acceptable

Reply:
just add a `setup_class` to EthApiTestCase to create a transaction there, as result it will be done once per test class and the data will remain consistent

it's not great to have so many assertions one by one cause if a single of those fails others won't be executed at all

Reply:
So, `setup_class` didn't work for whatever reason, so I ended up having to learn what a fixture is
![image](https://github.com/user-attachments/assets/03ec9f47-e7ec-4454-8167-4e55937096be)

Reply:
anyway, the test is split into different functions now and the Tx data is only generated once, which was my intention.